### PR TITLE
Add cancelled bookings report

### PR DESCRIPTION
**Pull request by konstyefremov**
https://github.com/happy-travel/agent-app-project/issues/409

### CODE REVIEW

File: Api/Models/Reports/CancelledBookingsData.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+using System;
+
+namespace HappyTravel.Edo.Api.Models.Reports
+{
+    public readonly struct CancelledBookingsData
```
Comment (kirillta) [inline Api/Models/Reports/CancelledBookingsData.cs:5]: CancelledBookingsReportData

File: Api/Controllers/AdministratorControllers/ReportsController.cs [hunk 2]
```diff
@@ -243,6 +243,26 @@ public async Task<IActionResult> GetVccBookingReport(DateTime from, DateTime end
...
+        [ProducesResponseType(typeof(FileStream), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        [AdministratorPermissions(AdministratorPermissions.BookingReportGeneration)]
+        public async Task<IActionResult> GetCancelledBookingsReport(DateTime from, DateTime end)
+        {
+            var (_, isFailure, stream, error) = await _reportService.GetCancelledBookingsReport(from, end);
+            if (isFailure)
+                return BadRequest(ProblemDetailsBuilder.Build(error));
+
+            return new FileStreamResult(stream, new MediaTypeHeaderValue("text/csv"))
```
Comment (kirillta) [inline Api/Controllers/AdministratorControllers/ReportsController.cs:259]: We have this code sequence everywhere across the class. Could it be refactored to a method?

**Review by kirillta — APPROVED** [review]

File: Api/Models/Reports/CancelledBookingsData.cs (hunk 1 above)
Comment (konstyefremov) [reply Api/Models/Reports/CancelledBookingsData.cs:5]: Done

File: Api/Controllers/AdministratorControllers/ReportsController.cs (hunk 2 above)
Comment (konstyefremov) [reply Api/Controllers/AdministratorControllers/ReportsController.cs:259]: Will resolve separately after merging the last PR's

**Pull request merged**